### PR TITLE
feat: support `/api/v1/label/<label_name>/values` from Prometheus

### LINKED ISSUE #1542: Support `/api/v1/label/<label_name>/values` from Prometheus
### What problem does the new feature solve?

The interface [`/api/v1/label/<label_name>/values`](https://prometheus.io/docs/prometheus/latest/querying/api/#querying-label-values) is for querying label values from the dataset. It can be translated to something like `SELECT * FROM t GROUP BY (label_name)` in SQL

Related to #1016

### What does the feature do?

Implement the interface [`/api/v1/label/<label_name>/values`](https://prometheus.io/docs/prometheus/latest/querying/api/#querying-label-values). Notice that the interface only accept `GET` requests.

### Implementation challenges

- Implement the HTTP endpoint
- translate query to SQL
- transform SQL result to Prometheus' format

### PR DESCRIPTION
**Pull request by gitccl**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Support `/api/v1/label/<label_name>/values` from Prometheus

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [X]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

close #1542

### CODE REVIEW

**Review by waynexia — APPROVED** [review]
looks good to me overall

File: src/servers/src/prom.rs [hunk 1]
```diff
@@ -699,3 +704,115 @@ fn promql_expr_to_metric_name(expr: &PromqlExpr) -> Option<String> {
...
+
+    let mut label_values = HashSet::new();
+
+    for query in queries {
+        let prom_query = PromQuery {
+            query,
+            start: start.clone(),
+            end: end.clone(),
+            // TODO(ccl): find a better value for step
+            step: DEFAULT_LOOKBACK_STRING.to_string(),
```
Comment (killme2008) [inline src/servers/src/prom.rs:742]: Looks like the `labels` API doesn't support `step`, so the `TODO` is unnecessary.

File: src/servers/src/prom.rs [hunk 2]
```diff
@@ -699,3 +704,115 @@ fn promql_expr_to_metric_name(expr: &PromqlExpr) -> Option<String> {
...
+) -> Result<()> {
+    let Some(label_col_idx) = batches.schema().column_index_by_name(label_name) else {
+        return Ok(());
+    };
+
+    // check whether label_name belongs to tag column
+    match batches
+        .schema()
+        .column_schema_by_name(label_name)
+        .unwrap()
```
Comment (killme2008) [inline src/servers/src/prom.rs:796]: The `unwrap` is not safe here, may panic.

File: src/servers/src/prom.rs [hunk 3]
```diff
@@ -699,3 +704,115 @@ fn promql_expr_to_metric_name(expr: &PromqlExpr) -> Option<String> {
...
+        .data_type
+    {
+        ConcreteDataType::String(_) => {}
+        _ => return Ok(()),
+    }
+
+    for batch in batches.iter() {
```
Comment (killme2008) [inline src/servers/src/prom.rs:803]: Looks like we don't have a better way to only query the label values specified by label name, but query all labels and filter the result.

cc @waynexia

File: src/servers/src/prom.rs (hunk 1 above)
Comment (gitccl) [reply src/servers/src/prom.rs:742]: But `PromQuery` requires `step`, different `step` may produce different query result?

File: src/servers/src/prom.rs (hunk 2 above)
Comment (gitccl) [reply src/servers/src/prom.rs:796]: The previous check ensures that the column is present:
```
  let Some(label_col_idx) = batches.schema().column_index_by_name(label_name) else {
      return Ok(());
  };
```

File: src/servers/src/prom.rs (hunk 1 above)
Comment (killme2008) [reply src/servers/src/prom.rs:742]: I mean we can always use the `DEFAULT_LOOKBACK_STRING`. The `TODO` comment is unnecessary.

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**